janusgraph-test/src/test/java/org/janusgraph/core/attribute/JtsGeoshapeHelperTest.java

```diff
@@ -42,4 +51,35 @@ public boolean supportJts() {
     public ShapeFactory getShapeFactory() {
         return new JtsShapeFactory((JtsSpatialContext) getHelper().context, (JtsSpatialContextFactory) getHelper().factory);
     }
+
+    @Test
+    public void testPolygon() {
```

Comment:
Is my understanding correct that this would not have been valid in the original code (which checked for object equality, rather than array equality), and hence would have thrown an exception?

Reply:
Actually，the original code may be just a slip of pen. It does nothing at most of the time or may throw an unexpected exception just when the first and last points are the same object (in address). It can be seen that we want a closed polygon (whose first and last point must be equal). I think we should test equality instead of inequality. Second, the symbol == or object equality you mentioned actually compares the address of the objects instead of their values. So I think here array equality is more suitable.